### ISSUE TITLE
Startup Help File Needed

### ISSUE THREAD

**Issue by nrcionline** [NTaylorMullen/ShootR#54]
I've downloaded and compiled https://github.com/BrennanConroy/ShootR
I have software developer experience, but I can't figure out how to start the program. It looks like the files need to be moved around and put in one place so they all work together.
I tried going to wwwroot/index.html but the screen only hangs there.
Does this mean it didn't compile or where do I start?
Do I need to start the server separately?

I used VS to start in debug mode and it got a bit farther. I see the ship now but it still hangs on the "loading" graphic (two-tone gray). Why is it not continuing? From VS it accesses the game by going to http://localhost:61691 - is this the standard way to start the game?

**Comment by NTaylorMullen** [comment]
@BrennanConroy could you share some guidance?

**Comment by BrennanConroy** [comment]
Looks like the current code doesn't work, I'll push a quick fix to my fork.

**Comment by BrennanConroy** [comment]
Should be good now 😄

**Comment by nrcionline** [comment]
Yes, that fixed it - at least when I run from VS. How do I start it outside of VS (what do I type into the browser?) The best I know would be to go to localhost/shootr where shootr points to the wwwroot, but this results in the same hanging (see picture).
![shootr](https://user-images.githubusercontent.com/32943620/53072389-fcb42c00-34b2-11e9-8d56-a2a30d9bd596.jpg)

**Comment by BrennanConroy** [comment]
Just try `http://localhost:<port>` you shouldn't need to navigate to a specific path

**Comment by nrcionline** [comment]
Okay. Which port? How do I define which port? Don't I have to get the program running for it to be listening on a port? The compiler only creates a .dll, not an executable. So, I'm confused as to how to start the program. Or if a website, don't I have to create an entry in IIS for the website and define a port? But then where do I point the website? All I've got is a default website that responds to the default port (80).

I found this file, but the urls don't work (I get a cannot connect which means nothing is there):
{
  "iisSettings": {
    "windowsAuthentication": false,
    "anonymousAuthentication": true,
    "iisExpress": {
      "applicationUrl": "http://localhost:64414/",
      "sslPort": 0
    }
  },
  "profiles": {
    "IIS Express": {
      "commandName": "IISExpress",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },
    "ShootR": {
      "commandName": "Project",
      "launchBrowser": true,
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      },
      "applicationUrl": "http://localhost:64415/"
    }
  }
}

**Comment by BrennanConroy** [comment]
When you dotnet run you should get some console output that says what port the app is listening on

**Comment by nrcionline** [comment]
That's helping me get closer. I've looked up the syntax for dotnet run, but isn't there a bat file with the command and the necessary arguments in it? What parameters do I pass to dotnet run to get it to work?

**Comment by BrennanConroy** [comment]
All you should need to do is go to the folder with the .csproj and run `dotnet run`. Assuming you have dotnet installed https://dotnet.microsoft.com/download

**Comment by nrcionline** [comment]
I can see how that starts everything and it at least pulls up now. I didn't change any code, but now it is hanging like it was before you made your fix. It hangs whether I use VS to start it, or I start from the command line with dotnet run. I've looked at main.js and it has your changes in it still and I've rebuilt the code several times. Still hangs. Let me know if you have any ideas. Do I need to clean out old files?

**Comment by nrcionline** [comment]
Apparently a firefox caching problem. I can run it on Edge and Chrome.
Thanks for your help!
